[quant] Add quantized::sigmoid that take output_scale/output_zero_point as input

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45920 [quant][fx][graphmode][api] Change API for custom module
* #45919 [quant][fx][graphmode] Run symbolic_trace in quantization
* #45885 [quant][fx][graphmode] Add support for sigmoid
* #45884 [quant][eagermode] Add support for sigmoid module
* #45883 [quant] Add quantized Sigmoid module
* **#45882 [quant] Add quantized::sigmoid that take output_scale/output_zero_point as input**

Summary:
Same changes as the stack for leaky_relu: https://github.com/pytorch/pytorch/pull/45702

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24129113](https://our.internmc.facebook.com/intern/diff/D24129113)

jerryzh168 committed:
[quant] Add quantized::sigmoid that take output_scale/output_zero_point as input

Summary:
Same changes as the stack for leaky_relu: https://github.com/pytorch/pytorch/pull/45702

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on aten/src/ATen/native/quantized/cpu/qsigmoid.cpp:
```diff
@@ -76,6 +73,15 @@ Tensor qnnpack_sigmoid(Tensor input) {
+
+// This ALWAYS outputs scale=1.0/256, dtype=quint8
+// The zero_point is 0 for qint32 and quint8, but -128 for qint8.
+Tensor qnnpack_sigmoid(Tensor input) {
```
Do we need this additional function? Can we directly call `qnnpack_sigmoid_observed_output` for qnnpack?

jerryzh168 replied:
oh sure

jerryzh168 committed:
Update on "[quant] Add quantized::sigmoid that take output_scale/output_zero_point as input"


Summary:
Same changes as the stack for leaky_relu: https://github.com/pytorch/pytorch/pull/45702

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24129113](https://our.internmc.facebook.com/intern/diff/D24129113)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Add quantized::sigmoid that take output_scale/output_zero_point as input"


Summary:
Same changes as the stack for leaky_relu: https://github.com/pytorch/pytorch/pull/45702

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24129113](https://our.internmc.facebook.com/intern/diff/D24129113)

[ghstack-poisoned]
